File: solution/java/week05-complexity/exercise2/src/main/java/command/GreetCommand.java
package command;

class GreetCommand implements Command {
    private final String firstName;
    private final String lastName;

    public GreetCommand(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    @Override
    public String executeAndDisplayResult() {
        return "Hello, " + firstName + " " + lastName + "!";
    }
}

File: solution/ts/week05-complexity/exercise2/src/greetCommand.ts
import {Command} from "./command";

export class GreetCommand implements Command {
    private readonly firstName: string;
    private readonly lastName: string;

    constructor(firstName: string, lastName: string) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    executeAndDisplayResult(): string {
        return `Hello, ${this.firstName} ${this.lastName}!`;
    }
}